dspace-server-webapp/src/main/java/org/dspace/app/rest/SubmissionCCLicenseSearchController.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.ServletRequest;
+
+import org.apache.commons.lang3.StringUtils;
+import org.dspace.app.rest.converter.ConverterService;
+import org.dspace.app.rest.exception.DSpaceBadRequestException;
+import org.dspace.app.rest.model.SubmissionCCLicenseRest;
+import org.dspace.app.rest.model.SubmissionCCLicenseUrlRest;
+import org.dspace.app.rest.model.hateoas.SubmissionCCLicenseUrlResource;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.license.service.CreativeCommonsService;
+import org.dspace.services.RequestService;
+import org.dspace.utils.DSpace;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This controller is responsible for searching the CC License URI
+ */
+@RestController
+@RequestMapping("/api/" + SubmissionCCLicenseRest.CATEGORY + "/" + SubmissionCCLicenseRest.PLURAL + "/search" +
+        "/rightsByQuestions")
+@PreAuthorize("permitAll()")
```

Comment:
@tdonohue You are correct that this one can move to a RestRepository. However there a couple of options. The method in this controller returns a `SubmissionCCLicenseUrlRest` while the `SubmissionCCLicenseRestRepository` supports `SubmissionCCLicenseRest ` objects. So our options are the following:

- Move to `SubmissionCCLicenseRestRepository` but then we break the mold that each repository supports a single rest object (didn't find a repo that supports multiple)
- Create a new `SubmissionCCLicenseUrlRestRepository` just to support the search method, the findOne & findAll() would throw not implemented exceptions. Downside here is that we have a lot more code than we have in the controller
- We leave it in the separate controller

Please let us know which option you prefer.

Reply:
@KevinVdV .  This should be in a separate repository `SubmissionCCLicenseUrlRestRepository`.  You are correct, we shouldn't ever have one Repository return different objects/resources.  

It's also worth noting, this Controller is incorrectly designed as-is for another reason you just pointed out. The Controller is wrongly using the `/api/configuration/submissioncclicense` resource path to represent a *different resource*.  So, in REST, a resource path should always represent the same type of object, and I'm now realizing from your comments that we have this incorrect behavior:

* `/api/config/submissioncclicense` root path represents a `SubmissionCCLicenseResource` object
* However, we're wrongly setting up `/api/config/submissioncclicense/search/rightsByQuestions` to represent a *different*, similar object called `SubmissionCCLicenseUrlResource`. 

Since these are *different* resources, they **must be using different paths**.  That means our Contract needs updating to have two related paths:
* `/api/config/submissioncclicense` - which already exists
* `/api/config/submissioncclicenseurl/search/rightsByQuestions` (or some similar path) which returns a URL resource when searching via questions. This would require updating this new contract PR: https://github.com/DSpace/Rest7Contract/pull/129

It'd be important to document in the Contract how these two types of resources are used together. It looks like the expectation here is to use `/submissioncclicense` to get a list of license options, then pass one license ID (with optional answers) to `/submissioncclicenseurl` to get the final URI.  If that's accurate, we should document that in the REST Contract to describe how clients are expected to interact with these endpoints

Reply:
@KevinVdV : Minor update.  I was just remembering we have an `/api/submission` path in our REST API.  So, since those `/api/config` names are getting really long, we could think about moving these to:
* `/api/config/submissioncclicense` -> rename `/api/submission/cclicense`
* `/api/config/submissioncclicenseurl` -> rename `/api/submission/cclicenseurl`

This isn't required out of this PR if you disagree / have concerns.  But, I wanted to note it here in case you agree these paths make more sense & are easier.   I do realize we have some submission configs under `/api/config` so this is fine as-is if you prefer it.  These endpoints are a little less about simply reading configuration, and more about looking up a CC license, so they could be moved to `/api/submission` instead.